HoloJS/ThreeJSApp/MyExtension.cpp

```diff
@@ -0,0 +1,44 @@
+#include "pch.h"
+#include "MyExtension.h"
+
+void MyExtension::Initialize() 
+{
+    JsValueRef globalObject;
+    JsGetGlobalObject(&globalObject);
+
+    JsPropertyIdRef myExtensionPropertyId;
+    JsGetPropertyIdFromName(L"MyExtension", &myExtensionPropertyId);
+
+    JsValueRef myExtensionProperty;
+    JsCreateObject(&myExtensionProperty);
+    JsSetProperty(globalObject, myExtensionPropertyId, myExtensionProperty, true);
+
+    JsPropertyIdRef addPropertyId;
+    JsGetPropertyIdFromName(L"add", &addPropertyId);
+    JsCreateFunction(add, nullptr, &m_add);
```

Comment:
Projecting functions and properties into JavaScript should be done through the HoloJsAppView object; otherwise an extension might inadvertently write over HoloJS required interfaces. HoloJsAppView should project all extension objects to a specific object in JavaScript, maybe global.holojs.extensions.

Reply:
Hmmm. As far as I can tell, it wouldn't be possible to restrict this? What are your thoughts on "polyfill" use cases?

Do you think it might be a good idea to have a HologramJs::ScriptingAPI which would provide a CLR wrapper class for Chakra that would provide commonly functionality like projecting, parsing of arguments and creating objects etc? It would most likely encourage best practice , would keep a consistent style of writing code, avoid the need for importing multiple headers, and probably make things easier to work with.

Reply:
In effect HoloJS already is a wrapper for Chakra, with implementations for graphics, input etc. I think it would be beyond the scope of the project to focus on a native extension model.

You're right, because of the WinRT limitations, it's not easy to expose creation of JS objects and functions over a WinRT interface. 

To make progress, how about you keep the "OnBeforeRun" event to HoloJS but don't modify the ThreeJS sample to use it yet?

Reply:
Removed the extension code and I believe the submodule issue has been fixed as well.